clients/venice-push-job/src/main/resources/avro/ValidateSchemaAndBuildDictOutput.avsc

```diff
@@ -0,0 +1,18 @@
+{
+  "name": "ValidateSchemaAndBuildDictOutput",
+  "namespace": "com.linkedin.venice.hadoop.output.avro",
+  "type": "record",
+  "doc": "The data protocol between ValidateSchemaAndBuildDictMapper and VenicePushJob",
+  "fields": [
+    {
+      "name": "key",
+      "doc": "The raw key bytes",
+      "type": "bytes"
+    },
+    {
+      "name": "value",
+      "doc": "The raw value bytes",
+      "type": "bytes"
+    }
+  ]
```

Comment:
Let's keep this as?
```
{
  "name": "ValidateSchemaAndBuildDictOutput",
  "namespace": "com.linkedin.venice.hadoop.output.avro",
  "type": "record",
  "doc": "The data protocol between ValidateSchemaAndBuildDictMapper and VenicePushJob",
  "fields": [
    {
      "name": "zstdDictionary",
      "doc": "The raw value bytes",
      "type": "bytes"
    }
  ]
}
```



The reasons for this are:
1. This is local to each VPJ instance - it is written and read by the same job
2. Putting key/value pairs as separate records means we need to check which keys are present and then do error handling. 
    1. This is not a problem now - but if we evolve later with complicated rules, we'll have to keep track of which keys we've seen, etc.
    2. By making the data as a single record, we can validate the rules in a single context.
3. By keeping `bytes` keys, we are essentially opening up to dumping anything and everything in there. We should be mindful about what we're persisting.

Reply:
Thanks for the suggestion @nisargthakkar. Fixed this.